File: src/main/java/me/itslucas/bookstore/beans/Book.java
package me.itslucas.bookstore.beans;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

@Entity
public class Book {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    private String name;
    private String author;
    private String description;
    private String type;
    private int quantity;
    private float price;
    private float discount;

    public Book(String name, String author, String description, String type, int quantity, float price, float discount) {
        this.name = name;
        this.author = author;
        this.description = description;
        this.type = type;
        this.quantity = quantity;
        this.price = price;
        this.discount = discount;
    }

    public Book() {

    }

    public void setId(Long id) {
        this.id = id;
    }


    public Long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getAuthor() {
        return author;
    }

    public String getDescription() {
        return description;
    }

    public String getType() {
        return type;
    }

    public int getQuantity() {
        return quantity;
    }

    public float getPrice() {
        return price;
    }

    public float getDiscount() {
        return discount;
    }
}

File: bookstore-core-service/src/main/kotlin/me/itslucas/bookstore/domain/Book.kt
package me.itslucas.bookstore.domain

import com.fasterxml.jackson.annotation.JsonIgnore
import org.springframework.web.multipart.MultipartFile
import javax.persistence.*

@Entity
class Book {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    var id: Long? = null
    var title: String? = null
    var author: String? = null
    var publisher: String? = null
    var publicationDate: String? = null
    var language: String? = null
    var category: String? = null
    var numberOfPages = 0
    var format: String? = null
    var isbn = 0
    var shippingWeight = 0.0
    var listPrice = 0.0
    var active = true

    @Column(columnDefinition = "text")
    var description: String? = null
    var inStockNumber = 0

    @Transient
    var bookImage: MultipartFile? = null

    @OneToMany(mappedBy = "book")
    @JsonIgnore
    var bookToCartItemList: List<BookToCartItem>? = null
}